nomad/job_endpoint_test.go

```diff
@@ -6858,87 +6873,103 @@ func TestJobEndpoint_Dispatch(t *testing.T) {
 
 			var dispatchResp structs.JobDispatchResponse
 			dispatchErr := msgpackrpc.CallWithCodec(codec, "Job.Dispatch", tc.dispatchReq, &dispatchResp)
+			if dispatchErr != nil {
+				must.True(t, tc.err)
+				must.ErrorContains(t, dispatchErr, tc.errStr)
+				return
+			} else if tc.err {
+				t.Logf("Expected error: %v, got: %+v", dispatchErr, dispatchResp)
+			}
 
-			if dispatchErr == nil {
-				if tc.err {
-					t.Fatalf("Expected error: %v", dispatchErr)
+			// Check that we got an eval and job id back
+			switch dispatchResp.EvalID {
+			case "":
+				must.Eq(t, tc.noEval, true)
```

Comment:
The existing test code has a similar issue to what I've pointed out above, where we're making an assertion on the test case field.